lbry/wallet/wallet.py

```diff
@@ -171,7 +172,12 @@ def pack(self, password):
     @classmethod
     def unpack(cls, password, encrypted):
         decrypted = better_aes_decrypt(password, encrypted)
-        decompressed = zlib.decompress(decrypted)
+        try:
+            decompressed = zlib.decompress(decrypted)
+        except zlib.error as e:
+            if "incorrect header check" in e.args[0].lower():
+                raise InvalidPasswordError()
+            raise
```

Comment:
is there a test for this?

Reply:
Added test steps to repeatedly call sync_apply() with a bad password.

Iterating 200 times is enough to detect the old failure most of the time. Sometimes it would take 500+ iterations (which is really expensive and could cause the case to timeout).